packages/airnode-deployer/src/infrastructure/index.test.ts

```diff
@@ -1521,3 +1534,288 @@ describe('deploymentInfo', () => {
+    expect(awsGetBucketDirectoryStructureSpy).toHaveBeenCalledTimes(1);
+    expect(awsGetBucketDirectoryStructureSpy).toHaveBeenCalledWith(bucket.name);
+    expect(awsGetFileFromBucketSpy).toHaveBeenCalledTimes(4);
+    expect(awsGetFileFromBucketSpy).toHaveBeenNthCalledWith(
+      1,
+      bucket.name,
+      '0xd0624E6C2C8A1DaEdE9Fa7E9C409167ed5F256c6/dev/1662558010204/config.json'
+    );
+    expect(awsGetFileFromBucketSpy).toHaveBeenNthCalledWith(
+      2,
+      bucket.name,
+      '0xd0624E6C2C8A1DaEdE9Fa7E9C409167ed5F256c6/dev/1662558010204/secrets.env'
+    );
+    expect(awsGetFileFromBucketSpy).toHaveBeenNthCalledWith(
+      3,
+      bucket.name,
+      '0xd0624E6C2C8A1DaEdE9Fa7E9C409167ed5F256c6/dev/1662558010204/config.json'
+    );
+    expect(awsGetFileFromBucketSpy).toHaveBeenNthCalledWith(
+      4,
+      bucket.name,
+      '0xd0624E6C2C8A1DaEdE9Fa7E9C409167ed5F256c6/dev/1662558010204/secrets.env'
+    );
+
+    const fetchedPath = path.join(tempOutputDirectory, fetchedFilename);
+    expect(() => fs.accessSync(fetchedPath, fs.constants.F_OK)).not.toThrow();
+    const zip = new AdmZip(fetchedPath);
+    zip.extractAllTo(tempOutputDirectory);
+    const configContent = fs.readFileSync(configPath);
+    const fetchedConfigContent = fs.readFileSync(path.join(tempOutputDirectory, 'config.json'));
+    expect(fetchedConfigContent).toEqual(configContent);
+    const secretsContent = fs.readFileSync(secretsPath);
+    const fetchedSecretsContent = fs.readFileSync(path.join(tempOutputDirectory, 'secrets.env'));
+    expect(fetchedSecretsContent).toEqual(secretsContent);
+  });
+
+  it('fetch files for a specific version of deployment', async () => {
+    const deploymentId = 'aws7195b548';
+    const deploymentVersion = '1f8210a2';
+    const fetchedFilename = `${deploymentId}-${deploymentVersion}.zip`;
+
+    const originalColorVariable = process.env.FORCE_COLOR;
+    // I have to disable table coloring so I can compare the output
+    process.env.FORCE_COLOR = '0';
+    await infrastructure.fetchFiles(deploymentId, tempOutputDirectory, deploymentVersion);
+    process.env.FORCE_COLOR = originalColorVariable;
+
+    expect(awsGetAirnodeBucketSpy).toHaveBeenCalledTimes(1);
+    expect(awsGetBucketDirectoryStructureSpy).toHaveBeenCalledTimes(1);
+    expect(awsGetBucketDirectoryStructureSpy).toHaveBeenCalledWith(bucket.name);
+    expect(awsGetFileFromBucketSpy).toHaveBeenCalledTimes(4);
+    expect(awsGetFileFromBucketSpy).toHaveBeenNthCalledWith(
+      1,
+      bucket.name,
+      '0xd0624E6C2C8A1DaEdE9Fa7E9C409167ed5F256c6/dev/1662558010204/config.json'
+    );
+    expect(awsGetFileFromBucketSpy).toHaveBeenNthCalledWith(
+      2,
+      bucket.name,
+      '0xd0624E6C2C8A1DaEdE9Fa7E9C409167ed5F256c6/dev/1662558010204/secrets.env'
+    );
+    expect(awsGetFileFromBucketSpy).toHaveBeenNthCalledWith(
+      3,
+      bucket.name,
+      '0xd0624E6C2C8A1DaEdE9Fa7E9C409167ed5F256c6/dev/1662557983568/config.json'
```

Comment:
Why is it called with different timestamps?

Reply:
The first two calls are searching for the right deployment. We need to read those files in order to generate the deployment ID and find the correct one. In this case, it's the first one. And the second two calls are actually fetching those files we need, those with the specific version (older ones).